packages/fxa-settings/src/pages/Signin/container.tsx

```diff
@@ -231,17 +229,20 @@ const SigninContainer = ({
 
   const beginSigninHandler: BeginSigninHandler = useCallback(
     async (email: string, password: string) => {
-      if (needsCanLinkAccountWebChannel) {
-        // If we're in the `force_auth` flow, users will not have answered
-        // "Confirm" to the Sync merge warning prompt yet, which is usually
-        // confirmed on email-first submission.
+      // If queryParamModel.hasLinkedAccount is defined, then we know the user
+      // came from email-first and was already prompted with the sync merge
+      // warning. The browser will automatically respond with { ok: true } without
+      // prompting the user if it matches the email the browser has data for.
+      if (
+        integration.isSync() &&
+        queryParamModel.hasLinkedAccount === undefined
+      ) {
         const { ok } = await firefox.fxaCanLinkAccount({ email });
         if (!ok) {
           return { data: undefined, error: undefined };
-        } else {
-          setNeedsCanLinkAccountWebChannel(false);
```

Comment:
We could also throw an error message here "User canceled login" or something along those lines.

Reply:
Good thinking, this should actually be easy because I can just return it as an `error`, lemme see if I can do that here real quick.